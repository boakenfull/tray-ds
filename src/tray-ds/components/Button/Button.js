import React from "react";
import { node, func, bool } from "prop-types";

import { ButtonStyled } from "./Button.styled";

const Button = ({ children, onClick, isIcon }) => (
  <ButtonStyled onClick={onClick} icon={isIcon} role="button">{children}</ButtonStyled>
);

Button.propTypes = {
  children: node.isRequired,
  onClick: func,
  isIcon: bool,
};

Button.defaultProps = {
  onClick: () => {},
  isIcon: false,
};

export default Button;
