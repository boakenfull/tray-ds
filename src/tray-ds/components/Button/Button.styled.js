import styled, { css } from "styled-components";
import { lighten } from "polished";

import { blueRibbon, white, comet } from "../../styles/colourPalette";
import { smallFontSize } from "../../styles/variables";

import { StyledIcon } from "../Icon/Icon.styled";

export const ButtonStyled = styled.button`
  font-family: "nunito-sans", sans-serif;
  font-weight: 600;
  background-color: ${blueRibbon};
  color: ${white};
  line-height: 40px;
  border-radius: 4px;
  font-size: ${smallFontSize};
  padding: 0 16px;
  outline: 0;

  &:hover {
    cursor: pointer;
    background-color: ${lighten(0.1, blueRibbon)};
  }

  &:active {
    background-color: ${lighten(0.15, blueRibbon)};
  }

  ${(props) => props.icon && css`
    border: 0;
    background-color: transparent;
    text-align: center;

    ${StyledIcon} {
      color: ${comet};
      display: unset;
    }

    &:hover {
      background-color: transparent;
    }
  `};
`;
