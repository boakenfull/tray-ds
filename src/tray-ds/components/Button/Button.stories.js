import React from "react";
import { text } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import Button from "./";

export default { title: "Button" };

export const Primary = () => <Button onClick={action("click")}>{text("Button Text", "Button")}</Button>;
