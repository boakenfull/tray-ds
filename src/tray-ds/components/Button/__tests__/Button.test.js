import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "../Button";

const handleOnClick = jest.fn();

test("renders button component", () => {
  const { container } = render(<Button onClick={() => handleOnClick()}>Button</Button>);
  expect(container).toMatchSnapshot();
});

test("handles onClick correctly", () => {
  const { getByText } = render(<Button onClick={() => handleOnClick()}>Button</Button>);
  fireEvent.click(getByText("Button"));

  expect(handleOnClick).toHaveBeenCalled();
});
