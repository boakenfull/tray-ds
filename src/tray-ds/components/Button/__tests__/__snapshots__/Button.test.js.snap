// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button component 1`] = `
<div>
  <button
    class="sc-AxirZ jQaZle"
    role="button"
  >
    Button
  </button>
</div>
`;
